File: ALittle/Module/ALittleIDE/Other/GameLibrary/CEngine/Script/CEngine/UISystem/Effect/EffectImage.lua

module("ALittle", package.seeall)

local ___pairs = pairs
local ___ipairs = ipairs


assert(ALittle.DisplayObject, " extends class:ALittle.DisplayObject is nil")
EffectImage = Lua.Class(ALittle.DisplayObject, "ALittle.EffectImage")

function EffectImage:Ctor(ctrl_sys)
end

function EffectImage:Action(control)
	if control == nil then
		return
	end
	self._show = __CPPAPIImage()
	self._texture = __CPPAPIRenderTexture()
	self.x = control.x
	self.y = control.y
	self.width = control.width
	self.height = control.height
	control._show:SetX(0)
	control._show:SetY(0)
	control.visible = true
	control.clip = false
	local result = self._texture:Draw(control._show, math.floor(self.width), math.floor(self.height), 1)
	control._show:SetX(self._x)
	control._show:SetY(self._y)
	if result then
		self._show:SetTexture(self._texture)
	end
	self._show:SetTextureCoord(0, 1, 0, 1)
	self.disabled = true
end

function EffectImage:Clear()
	if self._texture == nil then
		return
	end
	self._texture:Clear()
end

File: ALittle/Module/CEngine/JSScript/CEngine/UISystem/Effect/EffectImage.js
{
if (typeof ALittle === "undefined") window.ALittle = {};


if (ALittle.DisplayObject === undefined) throw new Error(" extends class:ALittle.DisplayObject is undefined");
ALittle.EffectImage = JavaScript.Class(ALittle.DisplayObject, {
	Ctor : function(ctrl_sys) {
	},
	Action : function(control) {
		if (control === undefined) {
			return;
		}
		this._show = ALittle.NewObject(lua.__CPPAPIImage);
		this._texture = ALittle.NewObject(lua.__CPPAPIRenderTexture);
		this.x = control.x;
		this.y = control.y;
		this.width = control.width;
		this.height = control.height;
		control._show.SetX(0);
		control._show.SetY(0);
		control.visible = true;
		control.clip = false;
		let result = this._texture.Draw(control._show, lua.math.floor(this.width), lua.math.floor(this.height), 1);
		control._show.SetX(this._x);
		control._show.SetY(this._y);
		if (result) {
			this._show.SetTexture(this._texture);
		}
		this._show.SetTextureCoord(0, 1, 0, 1);
		this.disabled = true;
	},
	Clear : function() {
		if (this._texture === undefined) {
			return;
		}
		this._texture.Clear();
	},
}, "ALittle.EffectImage");

}